src/sodaspark/scan.py

```diff
@@ -296,4 +303,136 @@ def execute(
     df.createOrReplaceTempView(scan_yml.table_name)
     scan = create_scan(scan_yml, soda_server_client=soda_server_client)
     scan.execute()
+    if as_frame:
+        return convert_scan_result_to_spark_data_frames(scan.scan_result)
     return scan.scan_result
+
+
+def measurements_to_data_frame(measurements: list[Measurement]) -> DataFrame:
+    """
+    Convert measurements to a data frame.
+    Parameters
+    ----------
+    measurements: List[Measurement]
+        The measurements.
+    Returns
+    -------
+    out : DataFrame
+        The measurements as data frame.
+    """
+    schema_group_values = T.StructType(
+        [
+            T.StructField(
+                "group", T.MapType(T.StringType(), T.StringType()), True
+            ),
+            T.StructField("value", T.StringType(), True),
+        ]
+    )
+    schema = T.StructType(
+        [
+            T.StructField("metric", T.StringType(), True),
+            T.StructField("column_name", T.StringType(), True),
+            T.StructField("value", T.StringType(), True),
+            T.StructField(
+                "group_values", T.ArrayType(schema_group_values, True), True
+            ),
+        ]
+    )
+    spark_session = SparkSession.builder.getOrCreate()
+    out = spark_session.createDataFrame(
+        measurements,
+        schema=schema,
+    )
+    return out
+
+
+def testresults_to_data_frame(testresults: list[TestResult]) -> DataFrame:
```

Comment:
Could you make `testresults` two words? `test_results`, in function name and parameter